utilities/cassandra/merge_operator.h

```diff
@@ -30,8 +32,16 @@ class CassandraValueMergeOperator : public MergeOperator {
 
  virtual bool AllowSingleOperand() const override { return true; }
 
+ virtual bool ShouldMerge(const std::vector<Slice>& operands) const override {
+   if (operands_limit_ > 0 && operands.size() >= operands_limit_) {
```

Comment:
```
virtual bool ShouldMerge(const std::vector<Slice>& operands) const override {
    return operands_limit_ == 0 || operands.size() <= operands_limit_;
}
```

Reply:
Thanks for the explanation. So the code should be like ^: If operands_limit_ is unlimited or operands.size() less or equal to limit, then we should merge.

Reply:
Oh `ShouldMerge` means we should stop looking further into other operands. Never mind my above comments. I though it means "should merge these operands"

Reply:
No, with the condition you provided above, it will be: invoke merge operator for any operands-vector-size less than the provided limit. 
Instead it should be: invoke merge operator when the operands vector size is greater than the limit (when limit > 0). I am using limit = 0 is disable this feature, by returning false. 
```
return operands_limit_ > 0 && operands.size() >= operands_limit_
```

May be the missing context that I should provide is: RocksDB will continue to merge operands as before if there are not enough operands to reach the limit even after exploring all the levels. `ShouldMerge` only helps in termination of the search for more operands for the key. 

Say, the limit is 4, and you only have 2 operands in all the levels combined, we will still invoke the merge operator with all 2 operands to return the right value for get. The current change is only for cases where the there are more than 4 merge operands, but don't want to go beyond looking more than 4 to keep tabs on tail latencies. If we continue to go deeper into the levels, we will incur higher read latencies due to doing more iops to fetch data from disk.  
During Rocksandra performance tuning, we found that P99 of merge operands is 8, so we wanted to see if putting a limit on merge operands can help.

Reply:
The above longer comment was return in response to your earlier suggested code change of: 
```
return operands_limit_ == 0 || operands.size() <= operands_limit_;
```
Seems like you already got the gist of it from your latest comment.

Reply:
may be `ShouldMerge` is a confusing name. I am open to renaming it to something which conveys the meaning much better.